spec/framework/config/config_reader_spec.rb

```diff
@@ -120,6 +120,23 @@
       end
     end
 
+    context 'with invalid config file' do
+      let(:config) { 'some corrupt information' }
+      let(:log) { instance_spy(Facter::Log) }
+
+      before do
+        allow(Facter::Log).to receive(:new).and_return(log)
+        allow(Hocon).to receive(:load).and_raise(StandardError)
+        allow(log).to receive(:warn)
+      end
+
+      it 'returns loggs a warning' do
+        config_reader.init
+
+        expect(log).to have_received(:warn)
```

Comment:
It would be good to also match the message, to make sure that is the codepath that you are testing.
```ruby
expect(log).to have_received(:warn).with(/Facter failed to read config file/)
```